app/code/community/Ebizmarts/MailChimp/Model/System/Config/Source/Account.php

```diff
@@ -39,52 +39,61 @@ class Ebizmarts_MailChimp_Model_System_Config_Source_Account
      */
     public function __construct()
     {
-        $helper = $this->helper = $this->getHelper();
+        $helper = $this->helper = $this->makeHelper();
         $scopeArray = explode('-', $helper->getScopeString());
 
         $mcStoreId = ($helper->getMCStoreId($scopeArray[1], $scopeArray[0])) ? $helper->getMCStoreId($scopeArray[1], $scopeArray[0]) : null;
         $listId = $helper->getGeneralList($scopeArray[1], $scopeArray[0]);
-        $api = $helper->getApi($scopeArray[1], $scopeArray[0]);
-        if ($api) {
-            try {
-                $this->_accountDetails = $api->root->info('account_name,total_subscribers');
-                if ($listId) {
-                    $listData = $api->lists->getLists($listId, 'stats');
-                    $this->_accountDetails['list_subscribers'] = $listData['stats']['member_count'];
-                }
-                if ($mcStoreId && $helper->getIfConfigExistsForScope(Ebizmarts_MailChimp_Model_Config::GENERAL_MCSTOREID, $scopeArray[1], $scopeArray[0])) {
-                    try {
-                        $storeData = $api->ecommerce->stores->get($mcStoreId, 'name,is_syncing');
-                        $this->_accountDetails['store_exists'] = true;
-                        $this->_accountDetails['store_name'] = $storeData['name'];
-                        //Keep both values for backward compatibility
-                        $this->_accountDetails['store_sync_flag'] = $storeData['is_syncing'];
-                        $this->_accountDetails['store_sync_date'] = $this->getDateSync($mcStoreId);
-                        $totalCustomers = $api->ecommerce->customers->getAll($mcStoreId, 'total_items');
-                        $this->_accountDetails['total_customers'] = $totalCustomers['total_items'];
-                        $totalProducts = $api->ecommerce->products->getAll($mcStoreId, 'total_items');
-                        $this->_accountDetails['total_products'] = $totalProducts['total_items'];
-                        $totalOrders = $api->ecommerce->orders->getAll($mcStoreId, 'total_items');
-                        $this->_accountDetails['total_orders'] = $totalOrders['total_items'];
-                        $totalCarts = $api->ecommerce->carts->getAll($mcStoreId, 'total_items');
-                        $this->_accountDetails['total_carts'] = $totalCarts['total_items'];
-                    } catch (MailChimp_Error $e) {
-                        $helper->deleteLocalMCStoreData($mcStoreId, $scopeArray[1], $scopeArray[0]);
-                        if ($listId) {
-                            $helper->createStore($listId, $scopeArray[1], $scopeArray[0]);
-                            $message = $helper->__('Looks like your MailChimp store was deleted. A new one has been created.');
-                            Mage::getSingleton('adminhtml/session')->addWarning($message);
+        try {
```

Comment:
Move the `try/catch` for `getApi` to a separate block (like in app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php line 249), to avoid nested `try/catch` blocks.